src/authorizations/users.rs

```diff
@@ -1,31 +1,78 @@
-use crate::config::Config;
+use reqwest::blocking::Client as ReqwestClient;
+use serde::Deserialize;
+use std::error::Error;
 
-pub fn is_authorized(id: &str, config: &Config) -> bool {
-    config.authorized_users.contains(id)
+// Checks for permissions in https://github.com/rust-lang/team/
+
+#[derive(Debug, Deserialize, Clone)]
+struct TeamResponse {
+    discord_ids: Vec<usize>,
+}
+
+#[derive(Debug)]
+struct TeamClient {
+    client: ReqwestClient,
+}
+
+impl TeamClient {
+    pub fn new() -> Self {
+        let team_client = ReqwestClient::new();
+
+        TeamClient {
+            client: team_client,
+        }
+    }
+}
+
+fn get_team_info() -> Result<TeamResponse, Box<dyn Error>> {
+    let team_client = TeamClient::new();
+
+    let team_request = team_client.client.get(
+        &String::from("https://team-api.infra.rust-lang.org/v1/permissions/crates_io_ops_bot.staging_crates_io.json")
+    );
+
+    let team_response = team_request.send().and_then(|res| res.error_for_status())?;
```

Comment:
```suggestion
    let team_response = team_request.send()?.error_for_status()?;
```

You can remove the `and_then` here.